config_defaults_inc.php

```diff
@@ -541,6 +541,20 @@
 $g_validate_email = ON;
 
 /**
+ * Enable support for logging in by email and password, in addition to
+ * username and password.  This will only work as long as there is a single
+ * user with the specified email address and the email address is not blank.
+ *
+ * @global integer $g_email_login_enabled
+ */
+$g_email_login_enabled = ON;
+
```

Comment:
Users re-use same email addresses and logins across the internet.  Hence, I would argue it is not less secure.  As for the load, I've added a check that the provided email is a valid email address before looking up the DB for emails.